torch/csrc/jit/mobile/nnc/aot_compiler.cpp

```diff
@@ -178,40 +183,184 @@ std::pair<std::unique_ptr<Function>, const std::string> aotCompile(
+    for (const auto& s : input_dims_str) {
+      input_dims.push_back(c10::stoi(s));
+    }
+    inputs.push_back(input_dims);
+  }
+  return inputs;
+}
+
+std::vector<at::ScalarType> parseInputTypes(
+    const std::string& input_types_str) {
+  std::vector<std::string> inputTypes = split(';', input_types_str);
+  std::vector<at::ScalarType> scalarTypes;
+  for (const auto& inputType : inputTypes) {
+    at::ScalarType scalarType;
+    if (inputType == "float") {
+      scalarType = at::ScalarType::Float;
+    } else if (inputType == "uint8") {
+      scalarType = at::ScalarType::Byte;
+    } else if (inputType == "int64") {
+      scalarType = at::ScalarType::Long;
+    } else {
+      CAFFE_THROW("Unsupported input type: ", inputType);
+    }
+    scalarTypes.push_back(scalarType);
+  }
+  return scalarTypes;
+}
+
+std::string getNncKernelId(
+    const std::string& model_name,
+    const std::string& model_version,
+    const std::string& method_name) {
+  // TODO: calculate the version_token.
+  const std::string version_token = "VERTOKEN";
+  return model_name + ":" + model_version + ":" + method_name + ":" +
+      version_token;
+}
+
+std::string getNncKernelFuncName(
+    const std::string& model_name,
+    const std::string& model_version,
+    const std::string& method_name) {
+  return "nnc_" + model_name + "_" + model_version + "_" + method_name;
+}
+
+std::shared_ptr<Graph> preprocessGraphPasses(
+    std::shared_ptr<Graph>& graph,
+    const std::vector<c10::optional<at::Tensor>>& example_inputs) {
+  GRAPH_DEBUG("Before preprocessing graph passes: ", *graph);
+  torch::jit::RemoveTensorMutation(graph);
+  torch::jit::EliminateDeadCode(graph->block());
+  graph = torch::jit::tensorexpr::removeUnusedSelfArgument(graph);
+
+  torch::jit::tensorexpr::annotateInputShapes(graph, example_inputs);
+  torch::jit::OptimizeFrozenGraph(graph, true);
+  torch::jit::PropagateShapesOnGraph(graph);
+  torch::jit::PeepholeOptimize(graph, false);
+  torch::jit::ConstantPropagation(graph);
+  torch::jit::PropagateShapesOnGraph(graph);
+  torch::jit::PeepholeOptimize(graph, false);
+  torch::jit::ConstantPropagation(graph);
+
+  tensorexpr::removeUnusedSelfArgument(graph);
 
   std::vector<at::IValue> example_values;
-  std::vector<c10::optional<at::Tensor>> example_inputs;
-  for (int i = 0; i < sizes.size(); ++i) {
-    auto example_input = at::rand(sizes[i]).to(at::dtype(types[i]));
-    example_values.emplace_back(example_input);
-    example_inputs.emplace_back(example_input);
+  example_values.reserve(example_inputs.size());
+  for (auto example_input : example_inputs) {
+    example_values.emplace_back(*example_input);
   }
+  graph = TraceGraph(graph, example_values);
+  tensorexpr::annotateInputShapes(graph, example_inputs);
 
-  GRAPH_DUMP("graph before compiler passes ", g);
-  tensorexpr::removeUnusedSelfArgument(g);
-  g = TraceGraph(g, example_values);
   // TODO: Remove annotateInputShapes pass when TraceGraph can also capture
```

Comment:
ultra nit: Have the comment with annotateInputShapes removing the empty line in between